### PR TITLE
CES-10396 - Bluestore support

### PR DESCRIPTION
**Pull request by gaelrehault**
Workaround for https://bugzilla.redhat.com/show_bug.cgi?id=1666822 to enable bluestore support for ceph 3.2

### CODE REVIEW

File: src/pilot/assign_role.py [hunk 1]
```diff
@@ -1049,53 +1076,101 @@ def get_drives(drac_client):
...
-            osd_config['devices'].append(osd_drive_device_name)
-            osd_config['dedicated_devices'].append(ssd_device_name)
+            mklvm.append('  device=$(ls -la ' + osd_drive_device_name + " |  awk -F \"../../\" '{ print $2 }')")
+            mklvm.append('  eval "wipefs -a /dev/${device}"')
+            mklvm.append('  sleep 2')
+            mklvm.append('  pvcreate ' + osd_drive_device_name)
+            mklvm.append('  vgcreate ceph_vg' + str(vg_index) + ' ' + osd_drive_device_name)
+            mklvm.append("  size=$(sudo fdisk -l /dev/${device} 2>/dev/null | grep -m1 \"Disk\" | awk '{print $5}')")
+            mklvm.append("  sizeb=`expr $((${size} * 99 / 100))`")
+            mklvm.append("  sizeb=`expr $((${sizeb} / 512 * 512))`")
```
Comment (jtwill98) [inline src/pilot/assign_role.py]: sizeb=`expr $((${sizeb} / 512 * 512))` -- isn't that always the sizeb?  why is it necessary yo divide and multiply with same number?

File: src/pilot/assign_role.py [hunk 2]
```diff
@@ -1501,11 +1576,10 @@ def main():
     except:  # noqa: E722
         LOG.exception("Unexpected error")
         sys.exit(1)
-    finally:
-        # Leave the node powered off.
-        if drac_client is not None:
+    finally:
```
Comment (jtwill98) [inline src/pilot/assign_role.py]: Is there tabs at the EOL?

File: src/pilot/assign_role.py [hunk 3]
```diff
@@ -911,17 +911,16 @@ def generate_osd_config(ip_mac_service_tag, drac_client):
...
-        new_osd_config = generate_osd_config_without_journals(controllers,
-                                                              ssds)
+        new_osd_config, mklvm = generate_osd_config_without_journals(controllers,
+                                                              ssds, system_id)
     elif len(ssds) > 0 and len(spinners) > 0:
         # If we have a mix of flash and spinners, then use the ssds as journals
-        new_osd_config = generate_osd_config_with_journals(controllers,
-                                                           spinners,
-                                                           ssds)
+        new_osd_config, mklvm = generate_osd_config_with_journals(controllers,
```
Comment (jtwill98) [inline src/pilot/assign_role.py]: I did not know you could do this... Does the variable assignments receive the same value?

File: src/pilot/assign_role.py (hunk 1 above)
Comment (gaelrehault) [reply src/pilot/assign_role.py]: The value needs to be rounded to the nearest 512 multiplier ( sector byte size ) for the volume to be created.

File: src/pilot/assign_role.py (hunk 3 above)
Comment (gaelrehault) [reply src/pilot/assign_role.py]: Yes you can, the methods now returns osd_config, mklvm

File: src/pilot/assign_role.py [hunk 4]
```diff
@@ -1049,53 +1076,101 @@ def get_drives(drac_client):
...
-        'devices': [],
-        'dedicated_devices': []}
+        'osd_scenario': 'lvm',
+        'osd_objectstore': 'bluestore',
+        'lvm_volumes': []}
     osd_index = 0
+    vg_index = 0
     remaining_ssds = len(ssds)
-    for ssd in ssds:
+    for ssd in ssds:
```
Comment (cdearborn) [inline src/pilot/assign_role.py]: There's some trailing whitespace in the line above

File: src/pilot/assign_role.py [hunk 5]
```diff
@@ -1501,11 +1576,10 @@ def main():
     except:  # noqa: E722
         LOG.exception("Unexpected error")
         sys.exit(1)
-    finally:
-        # Leave the node powered off.
-        if drac_client is not None:
+    finally:	
+        # Leave the node powered off.
```
Comment (cdearborn) [inline src/pilot/assign_role.py]: here too

File: src/pilot/assign_role.py [hunk 6]
```diff
@@ -1501,11 +1576,10 @@ def main():
     except:  # noqa: E722
         LOG.exception("Unexpected error")
         sys.exit(1)
-    finally:
-        # Leave the node powered off.
-        if drac_client is not None:
+    finally:	
+        # Leave the node powered off.	
+        if drac_client is not None:
```
Comment (cdearborn) [inline src/pilot/assign_role.py]: here too

**Pull request merged**